macros/models/_sector/add_amount_usd.sql

```diff
@@ -0,0 +1,58 @@
+{% macro add_amount_usd(
+    trades_cte
+    , prices_model
+    , bought_amount_column
+    , sold_amount_column
+) %}
+
+SELECT
+    bt.*
+    , CASE
+        WHEN bt.token_bought_address IN (
+                SELECT contract_address
+                FROM {{ ref('prices_trusted_tokens') }}
+                WHERE blockchain = bt.blockchain
+            )
+            THEN bt."{{ bought_amount_column }}" * pb.price
+        WHEN bt.token_sold_address IN (
+                SELECT contract_address
+                FROM {{ ref('prices_trusted_tokens') }}
+                WHERE blockchain = bt.blockchain
+            )
+            THEN bt."{{ sold_amount_column }}" * ps.price
```

Comment:
i'm thinking these `WHEN token IN (...)` statements are what is causing the slowdown in run times in CI. notice this last run took ~2000s, while it used to take ~200s (10x increase in runtime).
reference from curve CI:
https://github.com/duneanalytics/spellbook/actions/runs/7758815487/job/21161436499?pr=5271#step:11:68

if we change this logic to be `left join` to `prices_trusted_tokens`, similar to regular prices joins below, that may be more efficient. of course, we'd have to left join twice, once for bought, once for sold

Reply:
256s now:
https://github.com/duneanalytics/spellbook/actions/runs/7856500247/job/21439334222?pr=5130#step:11:20